### PR TITLE
Mp/server pagination

### PR DESCRIPTION
**Pull request by mvpoyatt**
## Summary by CodeRabbit

- **New Features**
  - Introduced dynamic `PacketsTable` component with column visibility, pagination controls, loading indicators, and a modal for row details.
  
- **Enhancements**
  - Improved search and filtering functionality for packets.
  - Enhanced table layout and style adjustments for better user experience.
  
- **Bug Fixes**
  - Fixed various styling issues in `chain-cell` and `Select` components.

- **Chores**
  - Updated internal interfaces and functions for better alignment with new features.

### CODE REVIEW

**Comment by mvpoyatt** [comment]
Screenshot 2024-07-12 at 4 16 07 PM

File: app/api/packets/helpers.ts [hunk 1]
```diff
@@ -211,3 +211,31 @@ export function generateQueryParams(params: QueryFilters): string {
...
+  let channelFilter = '';
+  if (srcFilter || destFilter) {
+    channelFilter = `sourceChannel: {AND: {${srcFilter}${destFilter}}}`;
+  }
+  if (channelFilter && (startFilter || endFilter || stateFilter)) {
+    channelFilter = ', ' + channelFilter
+  }
+
+  return startFilter + endFilter + stateFilter + channelFilter;
+}
```
Comment (Inkvi) [inline app/api/packets/helpers.ts:241]: This function is hard to reason about it. Can we simplify it?

Here are some ideas:
```
export function generateSendPacketFilters(filters: FiltersProps): string {
  const filterParts = [
    filters.start && `blockTimestamp_gte: ${filters.start}`,
    filters.end && `blockTimestamp_lte: ${filters.end}`,
    filters.states && `packetRelation: {state_in: ${filters.states}}`,
    (filters.src || filters.dest) && `sourceChannel: {AND: {
      ${filters.src ? `portId_contains: ${filters.src}` : ''}
      ${filters.src && filters.dest ? ',' : ''}
      ${filters.dest ? `counterpartyPortId_contains: ${filters.dest}` : ''}
    }}`
  ].filter(Boolean);

  return filterParts.join(', ');
}
```
or
```
export function generateSendPacketFilters(filters: FiltersProps): string {
  const startFilter = filters.start ? `blockTimestamp_gte: ${filters.start}` : '';
  const endFilter = filters.end ? `blockTimestamp_lte: ${filters.end}` : '';
  const stateFilter = filters.states ? `packetRelation: {state_in: ${filters.states}}` : '';
  const srcFilter = filters.src ? `portId_contains: ${filters.src}` : '';
  const destFilter = filters.dest ? `counterpartyPortId_contains: ${filters.dest}` : '';

  const channelFilter = (srcFilter || destFilter) 
    ? `sourceChannel: {AND: {${srcFilter}${srcFilter && destFilter ? ', ' : ''}${destFilter}}}`
    : '';

  return [
    startFilter, 
    endFilter, 
    stateFilter && `${startFilter || endFilter ? ', ' : ''}${stateFilter}`, 
    channelFilter && `${startFilter || endFilter || stateFilter ? ', ' : ''}${channelFilter}`
  ].filter(Boolean).join('');
}
```

File: app/api/packets/request-handlers.ts [hunk 2]
```diff
@@ -44,6 +48,41 @@ export async function getAllPackets(
...
+  src?: string,
+  dest?: string
+): Promise<PacketRes> {
+  const senderFilter = `OR: [{uchEventSender_eq: ${searchValue}}, {packetDataSender_eq: ${searchValue}}]`
+  const filterProps: FiltersProps = { start, end, states, src, dest };
+  const filters = generateSendPacketFilters(filterProps);
+
+  const queryParams = generateQueryParams({
+    orderBy: 'blockTimestamp_DESC',
+    where: 'AND: [{' + senderFilter + '}, {' + filters +' }]',
```
Comment (Inkvi) [inline app/api/packets/request-handlers.ts:67]: Using ` to embed vars into a string makes the code more readable

File: app/api/packets/route.ts [hunk 3]
```diff
@@ -11,35 +11,35 @@ export const dynamic = 'force-dynamic'; // defaults to auto
...
-  } else {
-    to = '';
+  if (searchValue) { searchValue = '"' + searchValue + '"'; }
+  if (start) { start = '"' + start + '"'; }
+  if (end) { end = '"' + end + '"'; }
+  if (src) { src = '"' + src + '"'; }
+  if (dest) { dest = '"' + dest + '"'; }
+  if (states) {
+    states = states.toUpperCase();
+    states = '[' + states + ']';
```
Comment (Inkvi) [inline app/api/packets/route.ts:31]: Can you use `` for string interpolation?

**Comment by Inkvi** [comment]
A couple of UX improvements that can be made:
- Use cursor:pointer in dropdown menus of the filters and the columns.
- Show the columns dropdown menu under the button.
- Each page change, causes a lot of UI rerendering. The table columns disappear and then appear in a short period of time causing a lot of jerking. Can we keep the column names in the table header while it's loading?

Screenshot 2024-07-15 at 10 38 21 AM
Screenshot 2024-07-15 at 10 38 34 AM
Screenshot 2024-07-15 at 10 39 28 AM

**Review by Inkvi — APPROVED** [review]
I'm approving it to unblock the next changes but cursor:pointer is still needed for the columns dropdown. The jerkiness is better but still present and needs to be addressed.

**Pull request merged**